sisl/physics/hamiltonian.py

```diff
@@ -505,7 +505,7 @@ def fermi_level(self, bz=None, q=None, distribution='fermi_dirac', q_tol=1e-10):
         bz.asarray()
 
         if q is None:
-            q = self.geometry.q0
+            q = self.geometry.q0 * 0.5 # Account for double occupancy
```

Comment:
This will not work for non-collinear and SOC calculations.

Instead I would do something like:
```python
if q is None:
    if self.is_unpolarized:
        q = self.geometry.q0 * 0.5
    else:
        q = self.geometry.q0
```
or?

Reply:
I think the above should be the only change, actually?

Reply:
True, this 'fix' only refers to strictly non-polarized Hamiltonians, where there is spin degeneracy. This change makes more sense probably, since is more specific.